src/google/protobuf/compiler/objectivec/objectivec_file.cc

```diff
@@ -168,32 +199,23 @@ void CollectMinimalFileDepsContainingExtensionsWorker(
 // There are comments about what the expected code should be line and limited
 // testing objectivec/Tests/GPBUnittestProtos2.m around compilation (#imports
 // specifically).
-void CollectMinimalFileDepsContainingExtensions(
-    const FileDescriptor* file,
-    std::vector<const FileDescriptor*>* files) {
-  std::set<const FileDescriptor*> files_visited;
-  for (int i = 0; i < file->dependency_count(); i++) {
-    const FileDescriptor* dep = file->dependency(i);
-    CollectMinimalFileDepsContainingExtensionsWorker(dep, files,
-                                                     &files_visited);
-  }
+const std::vector<const FileDescriptor*>
+FileGenerator::CommonState::CollectMinimalFileDepsContainingExtensions(
+    const FileDescriptor* file) {
+  std::set<const FileDescriptor*> min_deps =
```

Comment:
Are all the file descriptor names unique? Could you place the comparator on the sets, which would remove the need to sort the list afterwards? (You might even be able to return the set directly instead of copying it into a vector?)

Reply:
They should be unique or the other includes would quickly get confused.

I debated tweaking the compare, but in the case where we're only generating the top most file(s), we end up only having to do the sort(s) at the end when collecting the final result and generating vs. having to sort all the intermediate values also, so this seemed better from a performance pov.